python/mlx/nn/layers/linear.py

```diff
@@ -26,20 +41,92 @@ class Linear(Module):
 
     def __init__(self, input_dims: int, output_dims: int, bias: bool = True):
         super().__init__()
-        scale = math.sqrt(1 / input_dims)
+        self.input_dims = input_dims
+        self.output_dims = output_dims
+        self.weight = mx.zeros((output_dims, input_dims))
+        if bias:
+            self.bias = mx.zeros((output_dims,))
+
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        scale = math.sqrt(1.0 / self.input_dims)
         self.weight = mx.random.uniform(
             low=-scale,
             high=scale,
-            shape=(output_dims, input_dims),
+            shape=(self.output_dims, self.input_dims),
         )
+        if "bias" in self:
+            self.bias = mx.random.uniform(
+                low=-scale,
+                high=scale,
+                shape=(self.output_dims,),
+            )
```

Comment:
We should consider a more general solution to reseting parameters rather than just adding this to linear. For simplicity, I would remove it from this PR and if you are up for exploring a more general solution, let's put it into a follow on PR.

Reply:
Got it. I've reverted the changes for now, and we'll work on that in a future PR. https://github.com/ml-explore/mlx/pull/315/commits/a4b43db9af5a00285213d7c9ae710166663aca58.